src/tree/render.js

```diff
@@ -31,62 +28,33 @@ const filterTree = (id, nodeTree) => {
   });
 };
 
-const reRenderTree = ({ svg, divBox, activeNode, allNodes, o, width, height, cardStyling }) => {
+export const reRenderTree = ({
+  svg,
+  divBox,
+  activeNode,
+  allNodes,
+  o,
+  width,
+  height,
+  cardStyling,
+  callback,
+  storage,
+}) => {
   const nodes = filterTree(activeNode, allNodes);
-
-  const nodeIdList = nodes.map(node => node.data.id);
-  let appendNodes = [];
-
-  const activeNodes = svg.selectAll('.nodeWrapper');
-
-  // Cleanup from previous render
-  // eslint-disable-next-line no-underscore-dangle
-  if (previousNodes.length > 0 && activeNodes._groups[0].length > 0) {
-    const removeList = [];
-    previousNodes.forEach(node => {
-      if (nodeIdList.indexOf(node.data.id) === -1) {
-        removeList.push(node.data.id);
-      }
-    });
-    const previousNodesIdList = previousNodes.map(node => node.data.id);
-    nodes.forEach(node => {
-      if (previousNodesIdList.indexOf(node.data.id) === -1) {
-        appendNodes.push(node);
-      }
-    });
-    if (removeList.length > 0) {
-      svg
-        .selectAll('g')
-        .filter(data => removeList.indexOf(data.data.id) > -1)
-        // This inserts a transition for the removal of nodes. However it is not in sync with zooming at the moment
-        // .transition()
-        // .duration(transitionTime)
-        // .attr('transform', d => {
-        //   return `translate(${o.x(d.parent || d) + 150},${o.y(d.parent || d) + 100}) scale(0)`;
-        // })
-        .remove();
-
-      divBox
-        .selectAll('.node-rect')
-        .filter(data => removeList.indexOf(data.data.id) > -1)
-        .remove();
-    }
-  } else {
-    appendNodes = nodes;
-  }
-  previousNodes = nodes;
+  divBox.selectAll('.nodeRect').remove();
+  svg.selectAll('g').remove();
 
   // Create the nodes.
   const node = svg
     .selectAll('.node')
-    .data(appendNodes)
+    .data(nodes)
```

Comment:
Aren't we attaching also existing nodes again in this case?

Reply:
Wait we are deleting everything and building up everything again as well right?

Reply:
Yes, we are. I'm unsure if we need to pass the svg and divBox around. Since everything in them is removed, maybe create them inside rerender? @Caele thoughts?

Reply:
they only need to be created if the element changes really, so in theory they should only be created once, then we store a reference to them. 
Using the hooks we should be able to separate things into:
- loadData
- init elements
- construct tree
- render tree

This is fine for now though, it's the most straightforward way, not complicating things.